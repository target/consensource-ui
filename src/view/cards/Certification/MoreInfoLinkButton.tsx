import React from 'react';
import { CertResData } from 'services/api';
import { Button, makeStyles, createStyles } from '@material-ui/core';
import { Link } from 'view/components';

export interface MoreInfoLinkButtonProps {
  certificationId: CertResData['id'];
}

const useStyles = makeStyles(
  createStyles({
    align: {
      marginLeft: 'auto',
    },
  }),
);

export const MoreInfoLinkButton = ({
  certificationId,
}: MoreInfoLinkButtonProps) => {
  const classes = useStyles();

  return (
    <div className={classes.align}>
      <Link to={`/certifications/${certificationId}`}>
        <Button color="secondary" aria-label="More Info" title="More Info">
          More Info
        </Button>
      </Link>
    </div>
  );
};
