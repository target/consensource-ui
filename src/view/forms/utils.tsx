import React from 'react';
import { Grid, Typography, TextFieldProps } from '@material-ui/core';
import { BatchStatusRes } from 'services/api';

export interface TransactionFormProps {
  setBatchStatusLink: (statusLink: BatchStatusRes['link']) => void;
}

export type onChangeEvent = Parameters<
  NonNullable<TextFieldProps['onChange']>
>[0];

/**
 * Helper function that checks if any values in `state` have not been set.
 * Returns true if there is any key in `state` that is falsey.
 */
export function hasEmptyFields<T>(state: T) {
  return Object.keys(state).some((key) => !state[key as keyof T]);
}

export interface FormErrMsgProps {
  msg?: string;
}

export function FormErrMsg({ msg }: FormErrMsgProps) {
  if (!msg) {
    return null;
  }

  return (
    <Grid item>
      <Typography variant="caption" color="error">
        {msg}
      </Typography>
    </Grid>
  );
}
