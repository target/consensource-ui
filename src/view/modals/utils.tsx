import { DialogProps as MUIDialogProps } from '@material-ui/core';
import { FactoryResData } from 'services/api';

export interface DialogProps extends MUIDialogProps {
  handleClose: () => void;
}

export interface OrgDialogProps extends DialogProps {
  existing_org: FactoryResData;
}
