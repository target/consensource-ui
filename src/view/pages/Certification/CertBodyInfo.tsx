import React from 'react';
import { useQuery } from 'react-query';
import { Typography, Grid, CircularProgress } from '@material-ui/core';
import {
  LoadingWithMinDisplay,
  WarningIconError,
  InfoItem,
} from 'view/components';
import { fetchOrganizationById, CertResData } from 'services/api';

interface CertBodyInfoProps {
  certificate: CertResData;
}

export function CertBodyInfo({ certificate }: CertBodyInfoProps) {
  const { certifying_body_id, certifying_body } = certificate;
  const { isLoading, error, data } = useQuery('fetchOrganizationById', () =>
    fetchOrganizationById(certifying_body_id),
  );

  return (
    <LoadingWithMinDisplay
      isLoading={isLoading}
      loadingIndicator={<CircularProgress />}
    >
      {error && (
        <WarningIconError>Failed to load certifying body info</WarningIconError>
      )}
      {data && (
        <Grid container spacing={2}>
          <Grid item xs={12}>
            <Typography variant="h4">Certifying Body Info</Typography>
          </Grid>
          <InfoItem title="Certifying Body" val={certifying_body} />
        </Grid>
      )}
    </LoadingWithMinDisplay>
  );
}
