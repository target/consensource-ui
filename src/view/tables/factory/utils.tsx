import { MUIDataTableOptions, MUIDataTableChip } from 'mui-datatables';

export const DEFAULT_ROWS_PER_PAGE = 100;

export const textLabels: MUIDataTableOptions['textLabels'] = {
  body: {
    noMatch: 'No factories found',
  },
};

/**
 * Default props for the filter chips displayed in the
 * `<TableFilterList />`.
 */
export const filterChipProps: MUIDataTableChip = {
  color: 'secondary',
  variant: 'default',
};

/**
 * Utility function to convert a string to an array, or return
 * the original param if an array was passed.
 *
 * This operation is frequent when decoding URL query params.
 */
export const convertStrToArray = (val: string[] | string) => {
  return typeof val === 'string' ? [val] : val;
};
