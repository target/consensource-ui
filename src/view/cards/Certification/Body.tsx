import React from 'react';
import { Grid, Typography, makeStyles, createStyles } from '@material-ui/core';
import { CertResData } from 'services/api';
import { AssignmentTurnedInOutlined as CheckIcon } from '@material-ui/icons';
import { getLocaleFromUnix } from 'utils';
import {
  ClaimedIconButton,
  WarningIconError,
  UnclaimedIconButton,
} from 'view/components';

export interface BodyProps {
  validTo: CertResData['valid_to'];
  isClaimed: boolean;
}

export interface ValidToTextProps {
  validTo: BodyProps['validTo'];
}

export interface IsClaimedTextProps {
  isClaimed: BodyProps['isClaimed'];
}

const useStyles = makeStyles(({ palette }) =>
  createStyles({
    success: {
      color: palette.success.main,
    },
    warning: {
      color: palette.warning.main,
    },
    validToRow: {
      paddingLeft: 15,
    },
    claimedText: {
      paddingTop: 12.5,
    },
    validToIcon: {
      padding: '18px 16px 16px 16px !important',
    },
  }),
);

const ValidToText = ({ validTo }: ValidToTextProps) => {
  const classes = useStyles();

  const isValid = validTo >= Date.now();
  const text = isValid ? 'Valid until' : 'Expired on';

  return (
    <Grid container item spacing={4} className={classes.validToRow}>
      <Grid item xs={1} className={classes.validToIcon}>
        {isValid ? (
          <CheckIcon className={classes.success} />
        ) : (
          <WarningIconError />
        )}
      </Grid>
      <Grid item xs>
        <Typography variant="body1">{`${text} ${getLocaleFromUnix(
          validTo,
        )}`}</Typography>
      </Grid>
    </Grid>
  );
};

const IsClaimedText = ({ isClaimed }: IsClaimedTextProps) => {
  const classes = useStyles();

  return (
    <Grid container item spacing={4}>
      <Grid item xs={2}>
        {isClaimed ? <ClaimedIconButton /> : <UnclaimedIconButton />}
      </Grid>
      <Grid item xs>
        <Typography variant="body1" className={classes.claimedText}>
          {isClaimed ? 'Claimed' : 'Unclaimed'}
        </Typography>
      </Grid>
    </Grid>
  );
};

export const Body = ({ validTo, isClaimed }: BodyProps) => {
  return (
    <Grid container direction="column" spacing={1}>
      <Grid item>
        <Typography align="center" color="textSecondary">
          <i>Description</i>
        </Typography>
      </Grid>

      <Grid container item>
        <IsClaimedText isClaimed={isClaimed} />
      </Grid>

      <Grid container item>
        <ValidToText validTo={validTo} />
      </Grid>
    </Grid>
  );
};
