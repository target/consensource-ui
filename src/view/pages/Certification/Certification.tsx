import React from 'react';
import { useQuery } from 'react-query';
import {
  Typography,
  Divider,
  Grid,
  makeStyles,
  createStyles,
} from '@material-ui/core';
import {
  LoadingWithMinDisplay,
  FullScreenSpinnerWithLabel,
  WarningIconError,
} from 'view/components';
import { fetchCertificateById, CertResData } from 'services/api';
import { CertificateInfo } from './CertificateInfo';
import { CertBodyInfo } from './CertBodyInfo';

const useStyles = makeStyles(({ palette }) =>
  createStyles({
    check: {
      color: palette.success.main,
    },
    centeredText: {
      textAlign: 'center',
    },
  }),
);

export interface CertificationProps {
  certificationId: CertResData['id'];
}

export const Certification = ({ certificationId }: CertificationProps) => {
  const { isLoading, error, data } = useQuery('fetchCertificateById', () =>
    fetchCertificateById(certificationId),
  );

  const classes = useStyles();

  return (
    <LoadingWithMinDisplay
      isLoading={isLoading}
      loadingIndicator={
        <FullScreenSpinnerWithLabel label="Loading certificate info..." />
      }
    >
      {error && (
        <WarningIconError>Failed to load certification</WarningIconError>
      )}

      {data && (
        <Grid container spacing={6}>
          <Grid item xs={12}>
            <Typography className={classes.centeredText} variant="h3">
              {data.data.standard_name}
            </Typography>
            <Typography className={classes.centeredText} color="textSecondary">
              {`Granted to: ${data.data.factory_name}`}
            </Typography>
          </Grid>

          <Grid item xs={12}>
            <Divider variant="middle" />
          </Grid>

          <Grid item xs={12}>
            <CertificateInfo certificate={data.data} />
          </Grid>

          <Grid item xs={12}>
            <CertBodyInfo certificate={data.data} />
          </Grid>
        </Grid>
      )}
    </LoadingWithMinDisplay>
  );
};
