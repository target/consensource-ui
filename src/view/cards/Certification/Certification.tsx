import React, { useState } from 'react';
import {
  Card,
  CardActions,
  CardContent,
  Grid,
  makeStyles,
  createStyles,
} from '@material-ui/core';
import { CertResData } from 'services/api';
import { isDataClaimed } from 'utils';
import { ButtonLink } from 'view/components';
import { Header } from './Header';
import { Body } from './Body';

export interface CertificationCardProps {
  certification: CertResData;
}

const useStyles = makeStyles(
  createStyles({
    cardContent: { paddingBottom: 0 },
    moreInfoBtn: {
      marginLeft: 'auto',
    },
  }),
);

export const CertificationCard = ({
  certification,
}: CertificationCardProps) => {
  const classes = useStyles();
  const [isRaised, setIsRaised] = useState(false);

  return (
    <Card
      onMouseOver={() => setIsRaised(true)}
      onMouseOut={() => setIsRaised(false)}
      onFocus={() => setIsRaised(true)}
      onBlur={() => setIsRaised(false)}
      raised={isRaised}
    >
      <CardContent className={classes.cardContent}>
        <Grid container direction="column" spacing={2}>
          <Grid item>
            <Header standardName={certification.standard_name} />
          </Grid>

          <Grid item>
            <Body
              validTo={certification.valid_to}
              isClaimed={isDataClaimed(certification)}
            />
          </Grid>
        </Grid>
      </CardContent>

      <CardActions>
        <ButtonLink
          color="secondary"
          to={`/certifications/${certification.id}`}
          className={classes.moreInfoBtn}
        >
          More Info
        </ButtonLink>
      </CardActions>
    </Card>
  );
};
