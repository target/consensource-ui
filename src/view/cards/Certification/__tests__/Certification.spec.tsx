import userEvent from '@testing-library/user-event';
import React from 'react';
import { render, fireEvent, screen } from 'utils/testing';
import { mockCerts } from 'services/api/__mocks__';
import { CertificationCard } from '../Certification';

describe('<Certificates />', () => {
  const certificate = { ...mockCerts, data: [] as any };
  it('renders', () => {
    const { container } = render(
      <CertificationCard certification={certificate.data} />,
    );
    expect(container).toMatchSnapshot();
  });
  it('open up the cert page when more info link is clicked', () => {
    render(<CertificationCard certification={certificate.data} />);
    userEvent.click(screen.getByTitle('More Info'));
    /*expect(document.querySelector('More Info')?.getAttribute('href')).toBe(
      `{/certifications/${id}}`,
    );*/
  });
});
