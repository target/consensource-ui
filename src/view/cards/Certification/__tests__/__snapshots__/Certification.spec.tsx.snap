// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Certificates /> renders 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root makeStyles-cardContent"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <hr
                class="MuiDivider-root MuiDivider-middle"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
              >
                <i>
                  Description
                </i>
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-item"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-success"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm7 10c0 4.52-2.98 8.69-7 9.93-4.02-1.24-7-5.41-7-9.93V6.3l7-3.11 7 3.11V11zm-11.59.59L6 13l4 4 8-8-1.41-1.42L10 14.17z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1"
                  >
                    Claimed
                  </p>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-item"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorError"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1"
                  >
                    Expired on Invalid Date
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <div
        class="makeStyles-align"
      >
        <a
          class="makeStyles-link"
          href="/certifications/undefined"
        >
          <button
            aria-label="More Info"
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
            tabindex="0"
            title="More Info"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              More Info
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
